src/Metadata/Resource/Factory/AnnotationResourceMetadataFactory.php

```diff
@@ -111,8 +111,13 @@ private function createWith(ResourceMetadata $resourceMetadata, string $property
         $upperProperty = ucfirst($property);
         $getter = "get$upperProperty";
 
-        if (null !== $resourceMetadata->{$getter}()) {
-            return $resourceMetadata;
+        $currentValue = $resourceMetadata->{$getter}();
+        if (null !== $currentValue) {
+            if (null === $value) {
+                $value = $currentValue;
+            } elseif (is_array($currentValue)) {
+                $value = array_merge($currentValue, $value);
```

Comment:
Shouldn't it be recursive?

Reply:
Didn't notice that need TBH, only one level on my Sylius based project. 

Dumps were exactly like in: https://github.com/api-platform/core/pull/3711